syntax = "proto3";

package textprocessor;
option go_package = "server-b-chars/textprocessorpb";

// Servi√ßo para processamento de texto
service TextProcessor {
  // 1. Unary
  
  // Conta palavras (Servidor A)
  rpc CountWords (TextRequest) returns (WordCountResponse);

  // Conta caracteres (Servidor B)  
  rpc CountCharacters (TextRequest) returns (CharCountResponse);
}

// Mensagens
message TextRequest {
  string text = 1;
}

message WordCountResponse {
  int32 word_count = 1;
}

message CharCountResponse {
  int32 char_count = 1;
}